api/web3server_utils.go

```diff
@@ -0,0 +1,645 @@
+	}
+	ioAddress, err := address.FromBytes(common.HexToAddress(ethAddr).Bytes())
+	if err != nil {
+		return "", err
+	}
+	return ioAddress.String(), nil
+}
+
+func ioAddrToEthAddr(ioAddr string) (string, error) {
+	if len(ioAddr) == 0 {
+		return "0x0000000000000000000000000000000000000000", nil
+	}
+	addr, err := util.IoAddrToEvmAddr(ioAddr)
+	if err != nil {
+		return "", err
+	}
+	return addr.String(), nil
+}
+
+func uint64ToHex(val uint64) string {
+	return "0x" + strconv.FormatUint(val, 16)
+}
+
+func intStrToHex(str string) (string, error) {
+	amount, ok := big.NewInt(0).SetString(str, 10)
+	if !ok {
+		return "", errors.Wrapf(errUnkownType, "int: %s", str)
+	}
+	return "0x" + fmt.Sprintf("%x", amount), nil
+}
+
+func getStringFromArray(in interface{}, i int) (string, error) {
+	params, ok := in.([]interface{})
+	if !ok || i < 0 || i >= len(params) {
+		return "", errInvalidFormat
+	}
+	ret, ok := params[i].(string)
+	if !ok {
+		return "", errUnkownType
+	}
+	return ret, nil
+}
+
+func getStringAndBoolFromArray(in interface{}) (str string, b bool, err error) {
+	params, ok := in.([]interface{})
+	if !ok || len(params) != 2 {
+		err = errInvalidFormat
+		return
+	}
+	str, ok = params[0].(string)
+	if !ok {
+		err = errUnkownType
+		return
+	}
+	b, ok = params[1].(bool)
+	if !ok {
+		err = errUnkownType
+		return
+	}
+	return
+}
+
+func removeHexPrefix(hexStr string) string {
+	ret := strings.Replace(hexStr, "0x", "", -1)
+	ret = strings.Replace(ret, "0X", "", -1)
+	return ret
+}
+
+func (svr *Web3Server) getBlockWithTransactions(blkMeta *iotextypes.BlockMeta, isDetailed bool) (blockObject, error) {
+	transactionsRoot := "0x"
+	var transactions []interface{}
+	if blkMeta.Height > 0 {
+		actionInfos, err := svr.coreService.ActionsByBlock(blkMeta.Hash, 0, svr.coreService.cfg.API.RangeQueryLimit)
+		if err != nil {
+			return blockObject{}, err
+		}
+		for _, info := range actionInfos {
+			if isDetailed {
+				tx, err := svr.getTransactionFromActionInfo(info)
+				if err != nil {
+					log.L().Error("failed to get info from action", zap.Error(err), zap.String("info", fmt.Sprintf("%+v", info)))
+					continue
+				}
+				transactions = append(transactions, *tx)
+			} else {
+				transactions = append(transactions, "0x"+info.ActHash)
+			}
+		}
+		transactionsRoot = "0x" + blkMeta.TxRoot
+	}
+
+	if len(transactions) == 0 {
+		transactionsRoot = "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"
+	}
+	bloom := "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```

Comment:
in the above, i saw bloom = "0x0000" which has 0x prefix, here does not need prefix?